### PR TITLE
implement eslint and upgrade code style to ES6 standard

### PR DESCRIPTION
**Pull request by avahowell**
Addresses #232

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 **/*.js
 !plugins/Wallet/*
+!index.js
+!js/*
```
Comment (lukechampine) [inline .eslintignore:4]: so what's actually excluded at this point? Just the non-wallet plugins?

File: js/mainjs/addIPCListeners.js [hunk 2]
```diff
@@ -1,56 +1,53 @@
-'use strict';
-
 // Electron main process libraries
-const Electron = require('electron');
-const Dialog = Electron.dialog;
-const IPCMain = Electron.ipcMain;
-const Menu = Electron.Menu;
+import { dialog, ipcMain, Menu } from 'electron'
```
Comment (lukechampine) [inline js/mainjs/addIPCListeners.js:2]: what's the rule for capital vs lowercase?

File: .eslintignore (hunk 1 above)
Comment (avahowell) [inline .eslintignore:4]: Yes.

File: js/mainjs/addIPCListeners.js (hunk 2 above)
Comment (avahowell) [inline js/mainjs/addIPCListeners.js:2]: ES6 member imports (`import { blah ...}`) import exported members directly by name.  As such, they must match the names defined in the module.

File: .eslintignore (hunk 1 above)
Comment (lukechampine) [inline .eslintignore:4]: there's probably an easier way to specify that...but I'm assuming we won't need an `.eslintignore` soon anyway

**Pull request merged**